src/serializers/shared.rs

```diff
@@ -148,6 +149,29 @@ combined_serializer! {
     }
 }
 
+fn get_new_type(py: Python, type_: &str) -> PyResult<String> {
```

Comment:
To avoid allocating:

```suggestion
fn get_new_type<'a>(py: Python, type_: &'a str) -> PyResult<&'a str> {
```

(and then drop the `.to_owned()` calls below)